### PR TITLE
Add a helper function to decode scheduler plugin args

### PR DESCRIPTION
**Pull request by hex108**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup

> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

Most plugins need to decode their args. Then plugin authors need to figure out what is `runtime.Unkonw` and how to decode it. This PR adds a helper function `DecodeInto` to decode scheduler plugin args whose type is `runtime.Unkown`, it will be more convenient.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Add a helper function to decode scheduler plugin args.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

/sig scheduling

### CODE REVIEW

**Comment by hex108** [comment]
It will be useful for plugin developers to decode plugin's args. cc @bsalamat @ahg-g

File: pkg/scheduler/framework/v1alpha1/registry.go [hunk 1]
```diff
@@ -20,11 +20,36 @@ import (
...
+	}
+
+	var data []byte
+	switch configuration.ContentType {
+	// If ContentType is empty, it means ContentTypeJSON by default.
+	case runtime.ContentTypeJSON, "":
+		data = configuration.Raw
+	case runtime.ContentTypeYAML:
+		var err error
+		if data, err = yaml.ToJSON(configuration.Raw); err != nil {
```
Comment (alculquicondor) [inline pkg/scheduler/framework/v1alpha1/registry.go]: why not decode directly?

Comment (hex108) [reply pkg/scheduler/framework/v1alpha1/registry.go]: I think either one is OK. No strong tendency for it. Just follow the pattern  in https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/runtime/serializer/yaml/yaml.go#L40.

Comment (alculquicondor) [reply pkg/scheduler/framework/v1alpha1/registry.go]: TBH, I would change the code in apimachinery. It adds unnecessary processing.

Comment (hex108) [reply pkg/scheduler/framework/v1alpha1/registry.go]: I tried to find the answer, and found it at http://ghodss.com/2014/the-right-way-to-handle-yaml-in-golang/. And https://github.com/kubernetes-sigs/yaml exists for this reason, `this library first converts YAML to JSON using go-yaml and then uses json.Marshal and json.Unmarshal to convert to or from the struct. `.

Comment (hex108) [reply pkg/scheduler/framework/v1alpha1/registry.go]: It seems better to use `k8s.sigs.io/yaml` to unmashal yaml directly. Updated.

Comment (alculquicondor) [reply pkg/scheduler/framework/v1alpha1/registry.go]: Thanks. Much better to delegate this pattern to a single library.

**Review by wgliang — APPROVED** [review]

**Comment by hex108** [comment]
/assign @Huang-Wei 
for another review.

File: pkg/scheduler/framework/v1alpha1/registry_test.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+kind: KubeSchedulerConfiguration
+plugins:
+  permit:
+    enabled:
+      - name: foo
+pluginConfig:
+  - name: foo
+    args:
+      foo_test: "test decode"
+`,
```
[the remark this answers is not in the document]
Comment (hex108) [reply pkg/scheduler/framework/v1alpha1/registry_test.go]: YAML is a special case, if adding indentation, an error(`found character that cannot start any token`) will occur when decoding, so other places also declare a YAML string like this, e.g. https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/yaml/decoder_test.go#L34

[the remark this answers is not in the document]
Comment (hex108) [reply pkg/scheduler/framework/v1alpha1/registry_test.go]: Fixed the above one. PTAL

**Comment by Adirio** [comment]
Is this function expected to be used here @hex108: https://github.com/kubernetes/kubernetes/blob/b327a729c520ec6a8ba1d21ac86aff092d4d031a/pkg/scheduler/framework/plugins/examples/stateful/stateful.go#L69

Providing an example YAML or JSON in a comment in that example and modifying the example to use it would be useful.

**Comment by hex108** [comment]
> Is this function expected to be used here

@Adirio Yes, it is. I'll add it to one of examples. Thanks!

**Pull request merged**